File: src/KianoorTobi.Domain/Models/Product.cs
using KianoorTobi.Domain.Enums;

namespace KianoorTobi.Domain.Models
{
    public class Product : Entity
    {
        public string Name { get; set; }
        public decimal Price { get; set; }
        public ProductType Type { get; set; }
        public long ProductCategoryId { get; set; }

        public ProductCategory ProductCategory { get; set; }
    }
}

File: src/KianoorTobiFront/src/app/shared/models/product.ts
import { ProductType } from "../enums/product-type.enum";

export abstract class Product {
  name! :string
  price! :number;
  type! :ProductType;
  productCategoryId! :number;
  productCategoryName! :string;
}

export class ProductAddDto extends Product {
}

export class ProductEditDto extends Product {
  id! :number;
}

export class ProductOutputDto extends Product {
  id! :number;
}
